rust/kernel/module_param.rs

```diff
@@ -300,3 +323,152 @@ make_param_ops!(
     PARAM_OPS_BOOL,
     bool
 );
+
+/// An array of at __most__ `N` values.
+pub struct ArrayParam<T, const N: usize> {
+    values: [core::mem::MaybeUninit<T>; N],
+    used: usize,
+}
+
+impl<T, const N: usize> ArrayParam<T, { N }> {
+    fn values(&self) -> &[T] {
+        // SAFETY: `self` can only be generated and modified by the methods
+        // [`new`], and [`push`]. These maintain the invariant that the first
+        // `self.used` elements of `self.values` are initialized.
```

Comment:
This is a type invariant, right? In that case, it'd be nice to document it in `ArrayParam` directly, e.g.:

    /// # Invariant
    ///
    /// The first `self.used` elements of `self.values` are always initialized.

(matching `# Safety` sections)